cmd/private-org-peribolos-sync/main.go

```diff
@@ -0,0 +1,231 @@
+
+	"k8s.io/test-infra/prow/config/org"
+	"k8s.io/test-infra/prow/config/secret"
+	"k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/interrupts"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/config"
+	"github.com/openshift/ci-tools/pkg/promotion"
+)
+
+type gitHubClient interface {
+	GetRepo(owner, name string) (github.FullRepo, error)
+}
+
+type options struct {
+	peribolosConfig string
+	destOrg         string
+	releaseRepoPath string
+	whitelist       flagutil.Strings
+	github          flagutil.GitHubOptions
+
+	whitelistByOrg map[string]sets.String
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	fs.StringVar(&o.peribolosConfig, "peribolos-config", "", "Peribolos configuration file")
+	fs.StringVar(&o.releaseRepoPath, "release-repo-path", "", "Path to a openshift/release repository directory")
+	fs.StringVar(&o.destOrg, "destination-org", "", "Destination name of the peribolos configuration organzation")
+
+	fs.Var(&o.whitelist, "whitelist", "One or more repositories that do not promote official images and need to be included")
+
+	o.github.AddFlags(fs)
+	fs.Parse(os.Args[1:])
+
+	return o
+}
+
+func validateOptions(o *options) error {
+	var validationErrors []error
+
+	if len(o.releaseRepoPath) == 0 {
+		validationErrors = append(validationErrors, errors.New("--release-repo-path is not specified"))
+	}
+	if len(o.peribolosConfig) == 0 {
+		validationErrors = append(validationErrors, errors.New("--peribolos-config is not specified"))
+	}
+	if len(o.destOrg) == 0 {
+		validationErrors = append(validationErrors, errors.New("--destination-org is not specified"))
+	}
+
+	if len(o.whitelist.Strings()) > 0 {
+		if err := validateWhitelist(o.whitelist.Strings()); err != nil {
+			validationErrors = append(validationErrors, err)
+		} else {
+			o.whitelistByOrg = getWhitelistByOrg(o.whitelist.Strings())
+		}
+	}
+
+	if err := o.github.Validate(false); err != nil {
+		validationErrors = append(validationErrors, err)
+	}
+
+	return kerrors.NewAggregate(validationErrors)
+}
+
+func validateWhitelist(whitelist []string) error {
+	for _, repo := range whitelist {
+		if len(strings.Split(repo, "/")) != 2 {
+			return fmt.Errorf("--whitelist: %s must be in org/repo format", repo)
+		}
+	}
+	return nil
+}
+
+func getWhitelistByOrg(whitelist []string) map[string]sets.String {
+	ret := make(map[string]sets.String)
+
+	for _, repo := range whitelist {
+		orgRepo := strings.Split(repo, "/")
+
+		orgName := orgRepo[0]
+		repoName := orgRepo[1]
+
+		repos, exist := ret[orgName]
+		if !exist {
+			repos = sets.NewString()
+		}
+		ret[orgName] = repos.Insert(repoName)
+
+	}
+	return ret
+}
+
+func main() {
+	o := gatherOptions()
+	err := validateOptions(&o)
+	if err != nil {
+		logrus.WithError(err).Fatal("invalid options")
+	}
+	logger := logrus.WithField("destination-org", o.destOrg)
+
+	interrupts.OnInterrupt(func() {
```

Comment:
I do not understand why this is necessary at all. The _default_ behavior of a process that receives `SIGxxx` is to exit with an error code thats `128+SIGxxx`. Literally not having this code do _anything_ will get you the behavior you want, right?

Reply:
Exactly. But for some strange reason, the only way to stop the application from running is to `kill -9` the process. When it receives the signal there is log coming from github client that it receives it and it just continues running until it finishes.

Reply:
@stevekuznetsov AFAIK there's *something* (probalby in gh client by what @droslean writes?) in test-infra that causes the binaries to not simply exit on SIGINT et al, that's why we have these bits in our tools:

https://github.com/openshift/ci-tools/blob/d6f2ea08e5d0e8072549fde2bd87ca825a94df4a/cmd/cvp-trigger/main.go#L73-L76

https://github.com/openshift/ci-tools/blob/d6f2ea08e5d0e8072549fde2bd87ca825a94df4a/cmd/repo-init/main.go#L89-L92

I understand that we may be cargo-culting this snippet around, but without it the behavior is how Nikos describes it.

Reply:
@petr-muller no, the `repo-init` one is because I wanted to handle SIGINT while handling user input reads from the terminal. Let's dig.